packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -67,12 +67,42 @@ class DraggableScrollableController extends ChangeNotifier {
     return _attachedController!.extent.currentPixels;
   }
 
+  /// Get the attached sheet's current progress between the minimum extent and
+  /// the maximum extent of the sheet.
+  ///
+  /// If the sheet is at it's minimum extent, [progress] returns 0.0. If it is
+  /// at its max, it returns 1.0.
+  double get progress {
```

Comment:
`If you have access to a DraggableScrollableController, but do not have access to the corresponding DraggableScrollableSheet (eg because the sheet is held in an external package), you have no way of knowing the min and max child sizes of the attached sheet so you cannot tell how far between min and max the sheet currently is.`

Thanks for your contribute!
In this case, the external package should probably expose the min and max sizes to the developer rather than letting the framework support it. WDYT @caseycrogers

Reply:
I think having it embedded in the controller itself is very useful-requiring the package to expose it is tedium for the package maintainer and user of the package and means there is no longer a single source of truth.

But also I understand that Flutter has an interest in keeping it's code as minimal and maintainable as possible, so I guess this one could go either way-my personal preference is to merge this PR, but if you all feel it's unnecessary no worries if you close it!

Reply:
Yep! There is definitely no reason in theory the package can't expose the min/max this way.
However, it gets complicated in practice. A specific example would probably help:
https://github.com/ueman/feedback/blob/eb0dc79ba148325a9d2510ac4e8ddd752bdb992e/feedback/lib/src/feedback_bottom_sheet.dart#L85

Better feedback uses `DraggableScrollableSheet` and does expose the `minChildSize` via it's theme object. So it has an externally visible default value and you can configure that value. This gets much more complicated though as (looking at the line linked above) the effective `minChildSize` is actually different than the theme specified one as it's always in terms of the parent container's height. This is so that, when the keyboard opens, the sheet remains the same height rather than shrinking as the parent size shrinks.

There is probably a way to expose the effective `minChildSize` externally as a value listenable or something, but this is sufficiently inconvenient that I'd argue including the progress as a part of the controller is worth it for this and similar situations!

Reply:
Thanks for your expanation and that make sense.
It is sufficient to directly expose the min and max parameters so that the DSS does not become complicated by introducing new concepts.

Reply:
@caseycrogers hi, do you have planned to add the min and max property for the DSS controller?